pkg/util/test/util.go

```diff
@@ -3056,22 +3056,30 @@ func ValidateTelemetryV2Enabled(pxImageList map[string]string, cluster *corev1.S
 	}
 
 	// Validate Telemetry is Healthy in pxctl status
-	if err := validateTelemetryStatusInPxctl(true, cluster.Namespace); err != nil {
+	if err := validateTelemetryStatusInPxctl(true, cluster); err != nil {
 		return fmt.Errorf("failed to validate that Telemetry is Healthy in pxctl status, Err: %v", err)
 	}
 
 	logrus.Infof("All Telemetry components were successfully enabled/installed")
 	return nil
 }
 
-func validateTelemetryStatusInPxctl(telemetryShouldBeEnabled bool, namespace string) error {
+func validateTelemetryStatusInPxctl(telemetryShouldBeEnabled bool, cluster *corev1.StorageCluster) error {
 	listOptions := map[string]string{"name": "portworx"}
 	cmd := "pxctl status | grep Telemetry:"
 
+	if cluster.Spec.Security != nil {
```

Comment:
&& cluster.Spec.Security.Enabled
and move this inside of the retry block.

Reply:
Done